cmd/juju/controller/destroy.go

```diff
@@ -505,15 +507,12 @@ func (c *destroyCommand) getStorageAPI(modelName string) (storageAPI, error) {
 // SetFlags implements Command.SetFlags.
 func (c *destroyCommandBase) SetFlags(f *gnuflag.FlagSet) {
 	c.ControllerCommandBase.SetFlags(f)
-	f.BoolVar(&c.assumeYes, "y", false, "Do not ask for confirmation")
-	f.BoolVar(&c.assumeYes, "yes", false, "")
-	// This unused var is declared to pass a valid ptr into BoolVar
-	var noPromptHolder bool
-	f.BoolVar(&noPromptHolder, "no-prompt", false, "Does nothing. Option present for forward compatibility with Juju 3")
+	c.ConfirmationCommandBase.SetFlags(f)
 }
 
 // Init implements Command.Init.
 func (c *destroyCommandBase) Init(args []string) error {
+	c.ConfirmationCommandBase.Init(args)
```

Comment:
You need to check for error here

Imo `JUJU_SKIP_CONFIRMATION=gvhoeipd juju destroy-controller lxd-dev` should return the failed to parse error, but it's lost here

Reply:
Perhaps a unit test for cases like this would be useful

Reply:
Hope I fixed this...

Reply:
And added the unit-test